Source/Core/Core/AchievementManager.cpp

```diff
@@ -830,100 +327,106 @@ const AchievementManager::BadgeStatus& AchievementManager::GetGameBadge() const
   return m_game_badge;
 }
 
-const AchievementManager::UnlockStatus&
-AchievementManager::GetUnlockStatus(AchievementId achievement_id) const
+const AchievementManager::BadgeStatus& AchievementManager::GetAchievementBadge(AchievementId id,
+                                                                               bool locked) const
 {
-  return m_unlock_map.at(achievement_id);
+  if (locked)
+    return AchievementManager::GetInstance().m_locked_badges[id];
+  return AchievementManager::GetInstance().m_unlocked_badges[id];
 }
 
-AchievementManager::ResponseType
-AchievementManager::GetAchievementProgress(AchievementId achievement_id, u32* value, u32* target)
+const AchievementManager::LeaderboardStatus*
+AchievementManager::GetLeaderboardInfo(AchievementManager::AchievementId leaderboard_id)
```

Comment:
This function can probably be const as you aren't creating anything.  Additionally, it looks like there are multiple accesses into the map.

I would suggest doing something like this instead:

```
const AchievementManager::LeaderboardStatus*
AchievementManager::GetLeaderboardInfo(AchievementManager::AchievementId leaderboard_id) const
{
  if (const auto leaderboard_iter = m_leaderboard_map.find(leaderboard_id); leaderboard_iter != m_leaderboard_map.end())
  {
    if (leaderboard_iter->entries.size() == 0)
      FetchBoardInfo(leaderboard_id);
    return &*leaderboard_iter;
  }

  return nullptr;
}
```

Reply:
Yknow, I'm not sure I'd fully registered that this was legal if clause syntax before seeing this.

Reply:
Also, it's not const. FetchBoardInfo retrieves information that gets stored in m_client, so FetchBoardInfo is not const and neither is a method that calls it.

Reply:
Sorry, missed that `FetchBoardInfo` could modify, no need for the `const` then.